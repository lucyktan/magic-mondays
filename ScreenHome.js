import React, { Component } from 'react';
import {
  View,
  Text,
  Image,
  ImageBackground,
  StyleSheet,
  TouchableHighlight
} from 'react-native';
import { Container, Button} from 'native-base';

class ScreenHome extends Component {
  static navigationOptions = {
    header: null,
  };

  render() {
    const { navigate } = this.props.navigation;

    console.log(this.props, "props is here");
    return (
      <Container style={styles.backgroundImage}>
        <TouchableHighlight
          onPress={() => navigate("ScreenSplash", { screen: "Screen Splash" })}
          style={styles.magic}>
          <Image
            source={require('./img/app-home-with-promo.jpg')}
            style={styles.backgroundImage}
          />
        </TouchableHighlight>
      </Container>
    );
  }
};

const styles = StyleSheet.create({
  backgroundImage: {
    width: 414,
    height: 715,
    paddingTop: 5,
  },
  magic: {
    width: 600,
    height: 300,
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  button: {
    alignSelf: 'stretch',
    marginLeft: 10,
    marginRight: 10,
    borderRadius: 5,
    height: 40,
    justifyContent: 'center'
  },
  buttonText: {
    color: 'white',
    fontSize: 22,
    alignSelf: 'center'
  }
});
export default ScreenHome;