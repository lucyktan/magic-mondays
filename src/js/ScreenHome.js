import React, { Component } from 'react';
import {
  View,
  Text,
  Image,
  ImageBackground,
  StyleSheet,
  TouchableHighlight
} from 'react-native';
import { Container} from 'native-base';
import promoImage from '../img/app-home-with-promo.jpg';

class ScreenHome extends Component {

  render() {
    const { navigate } = this.props.navigation;

    console.log(this.props.navigation.state.routeName, "props is here");
    return (
      <Container style={styles.container}>
        <TouchableHighlight
          onPress={() => navigate("ScreenSplash", { screen: "Screen Splash" })}>
          <Image
            source={promoImage}
            style={styles.backgroundImage}
          />
        </TouchableHighlight>
      </Container>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '0%'
  },
  backgroundImage: {
    flex: 1,
    resizeMode: 'contain'
  }
});
export default ScreenHome;