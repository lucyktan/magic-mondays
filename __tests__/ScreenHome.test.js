import {View} from 'react-native';
import React from 'react';
import { configure, shallow } from 'enzyme';
import ScreenHome from '../src/js/ScreenHome';
import renderer from 'react-test-renderer';

describe('ScreenHome', () => {
  it('Page rendered', () => {

    let navigation = {
    state: {
        key: 'id-1512512594799-2',
        params: {
	      screen: 'Screen Home',
        user: 'prize'
  	  },
        routeName: 'ScreenHome',
      }
    };
    const rendered = shallow(
    	<ScreenHome navigation={navigation}/>
    );
    expect(rendered).toMatchSnapshot();
  });
});