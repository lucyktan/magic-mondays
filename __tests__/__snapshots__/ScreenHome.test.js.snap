// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenHome Page rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScreenHome
    navigation={
      Object {
        "state": Object {
          "key": "id-1512512594799-2",
          "params": Object {
            "screen": "Screen Home",
            "user": "prize",
          },
          "routeName": "ScreenHome",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableHighlight
        activeOpacity={0.85}
        onPress={[Function]}
        underlayColor="black"
      >
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
            }
          }
        />
      </TouchableHighlight>,
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": "0%",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.85,
        "children": <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
            }
          }
        />,
        "onPress": [Function],
        "underlayColor": "black",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": 1,
          "style": Object {
            "flex": 1,
            "resizeMode": "contain",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          underlayColor="black"
        >
          <Image
            source={1}
            style={
              Object {
                "flex": 1,
                "resizeMode": "contain",
              }
            }
          />
        </TouchableHighlight>,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": "0%",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.85,
          "children": <Image
            source={1}
            style={
              Object {
                "flex": 1,
                "resizeMode": "contain",
              }
            }
          />,
          "onPress": [Function],
          "underlayColor": "black",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "flex": 1,
              "resizeMode": "contain",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
